### PR TITLE
BUGFIX: TMA-1288 Users Brick - doesn't remove users from user groups

### PR DESCRIPTION
**Pull request by dangdinh**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1165/ : SUCCESS in 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1045/ : FAILURE in 11m 42s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1167/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1047/ : SUCCESS in 11m 40s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: lib/gooddata/models/project.rb [hunk 1]
```diff
@@ -1874,14 +1874,42 @@ def import_users(new_users, options = {})
...
       end
+
+      remove_member_from_group(users_lookup, removal_user_group_members)
       GoodData::Helpers.join(results, diff_results, [:user], [:login_uri])
     end
 
+    def remove_member_from_group(users_lookup, removal_user_group_members)
+      unless removal_user_group_members.empty?
+        users_lookup ||= login_users
+        current_user_groups = user_groups
```
[the remark this answers is not in the document]
Comment (dangdinh) [reply lib/gooddata/models/project.rb]: updated it

File: lib/gooddata/models/project.rb [hunk 2]
```diff
@@ -1874,14 +1874,42 @@ def import_users(new_users, options = {})
...
+        removal_user_group_members.uniq.each do |login|
+          user_uri = users_lookup[login]&.uri
+
+          # remove user from group if exists as group member
+          current_user_groups.each do |user_group|
+            if user_group.member?(user_uri)
+              GoodData.logger.warn("Removing #{user_uri} user from group #{user_group.name}")
```
[the remark this answers is not in the document]
Comment (dangdinh) [reply lib/gooddata/models/project.rb]: updated it

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1176/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1055/ : SUCCESS in 10m 47s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1195/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1072/ : SUCCESS in 12m 51s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1196/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1073/ : FAILURE in 12m 26s

**Comment by dangdinh** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1197/ : SUCCESS in 12s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1074/ : FAILURE in 12m 05s

**Comment by dangdinh** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/1198/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/1075/ : SUCCESS in 12m 03s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1874

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/94/ : SUCCESS in 5m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/636/ : SUCCESS in 23s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-gitops-deploy-pipeline/59/ : SUCCESS in 2m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/179/ : SUCCESS in 2m 11s (non-voting)

**Pull request merged**